internal/mapper/dashboard_mapper.go

```diff
@@ -0,0 +1,67 @@
+package mapper
+
+import (
+	"encoding/json"
+
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/giantswarm/observability-operator/pkg/domain/dashboard"
+)
+
+const grafanaOrganizationLabel = "observability.giantswarm.io/organization"
+
+// DashboardMapper handles conversion from Kubernetes resources to domain objects
+type DashboardMapper struct{}
+
+// New creates a new mapper
+func New() *DashboardMapper {
+	return &DashboardMapper{}
+}
+
+// FromConfigMap converts a Kubernetes ConfigMap to domain Dashboard objects
+func (m *DashboardMapper) FromConfigMap(cm *v1.ConfigMap) []*dashboard.Dashboard {
+	org := m.extractOrganization(cm)
+
+	var dashboards []*dashboard.Dashboard
+
+	for _, dashboardString := range cm.Data {
+		var content map[string]any
+		if err := json.Unmarshal([]byte(dashboardString), &content); err != nil {
+			// Create a dashboard with nil content for invalid JSON - let service layer handle validation
+			dash := dashboard.New("", org, nil)
+			dashboards = append(dashboards, dash)
```

Comment:
Why are we just ignoring the error here and creating an empty dashboard ?
Later validation would not be able to tell where is the invalid JSON coming from, here we can at least show at which index its coming from.

Reply:
Because the validation happens later on in the service for now